From 2dffef961db3b0506c50578227aca6f79582f869 Mon Sep 17 00:00:00 2001
From: Thierry Fauck <thiery@linux.vnet.ibm.com>
Date: Mon, 3 Jun 2019 09:56:18 +0200
Subject: [PATCH] With ppc64el arch and grub2 a primary partition of type Prep
 needs to be created - as this partition doesn't affect grub-pc processing, a
 new call to set the flag to the first partition is been created. Kernel
 release name for ppc64el arch is also added.

Signed-off-by: Thierry Fauck <thiery@linux.vnet.ibm.com>

	modified:   tools/autopkgtest-build-qemu
---
 tools/autopkgtest-build-qemu | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/tools/autopkgtest-build-qemu b/tools/autopkgtest-build-qemu
index 5b2c996..7b481d7 100755
--- a/tools/autopkgtest-build-qemu
+++ b/tools/autopkgtest-build-qemu
@@ -121,6 +121,9 @@ case "$mirror" in
       (ppc64)
         kernel=linux-image-powerpc64
         ;;
+      (ppc64el)
+        kernel=linux-image-powerpc64le
+        ;;
       (*)
         kernel="linux-image-$architecture"
         ;;
@@ -135,15 +138,24 @@ steps:
   - mkimg: "{{ image }}"
     size: 20G
 
-  - mklabel: msdos
+  - mklabel: gpt
     device: "{{ image }}"
 
   - mkpart: primary
     device: "{{ image }}"
     start: 0%
+    end: 5%
+    tag: boot
+
+  - mkpart: primary
+    device: "{{ image }}"
+    start: 5%
     end: 100%
     tag: root
 
+  - set: flag
+    device: "{{ image }}"
+
   - kpartx: "{{ image }}"
 
   - mkfs: ext4
-- 
2.7.4

